crates/jstzd/tests/octez_client_test.rs

```diff
@@ -351,3 +358,185 @@ async fn send_rollup_inbox_message() {
+    contract_path: &PathBuf,
+) -> ProtocolParameter {
+    let kernel = String::from_utf8(
+        tokio::fs::read(&installer_path)
+            .await
+            .unwrap_or_else(|e| panic!("failed to read installer file: {:?}", e)),
+    )
+    .unwrap();
+    let contract_json = serde_json::from_slice(
+        &tokio::fs::read(&contract_path)
+            .await
+            .unwrap_or_else(|e| panic!("failed to read contract file: {:?}", e)),
+    )
+    .unwrap();
+    let params = ProtocolParameterBuilder::new()
+        .set_bootstrap_smart_rollups([BootstrapSmartRollup::new(
+            rollup_address,
+            SmartRollupPvmKind::Wasm,
+            &kernel,
+            serde_json::json!({"prim": "bytes"}),
+        )
+        .unwrap()])
+        .set_bootstrap_contracts([BootstrapContract::new(
+            contract_json,
+            1_000_000,
+            Some(contract_address),
+        )
+        .unwrap()])
+        .set_source_path(
+            Path::new(std::env!("CARGO_MANIFEST_DIR"))
+                .join("tests/sandbox-params.json")
+                .to_str()
+                .unwrap(),
+        )
+        .build()
+        .unwrap();
+
+    let mut content = tokio::fs::read_to_string(params.parameter_file().path())
+        .await
+        .unwrap();
+    let mut value: serde_json::Value = serde_json::from_str(&content).unwrap();
+
+    // overwriting these config values so that outbox messages get cemented sooner
+    value.as_object_mut().unwrap().insert(
+        "smart_rollup_challenge_window_in_blocks".to_owned(),
+        serde_json::json!(8),
+    );
+    value.as_object_mut().unwrap().insert(
+        "smart_rollup_commitment_period_in_blocks".to_owned(),
+        serde_json::json!(8),
+    );
+
+    content = serde_json::to_string(&value).unwrap();
+    tokio::fs::File::create(params.parameter_file().path())
+        .await
+        .unwrap()
+        .write_all(content.as_bytes())
+        .await
+        .unwrap();
+
+    params
+}
+
+async fn wait_for_outbox_proof(rollup_rpc_endpoint: &str) -> anyhow::Result<OutputProof> {
+    #[derive(Deserialize)]
+    struct Message {
+        message_index: u32,
+    }
+    #[derive(Deserialize)]
+    struct Executable {
+        outbox_level: u32,
+        messages: Vec<Message>,
+    }
+
+    let url = format!("{rollup_rpc_endpoint}/local/outbox/pending/executable");
+    let (outbox_level, message_index) = poll(30, 1000, || async {
+        // response: [{"outbox_level": 1, "messages": [{"message_index": 0, ...}, {"message_index": 1, ...}]}]
+        if let Ok(res) = reqwest::get(&url).await {
+            let vs = res.json::<Vec<Executable>>().await.unwrap();
+            // using the first message here since any of those should work
+            if let Some(v) = vs.first() {
+                if let Some(m) = v.messages.first() {
+                    return Some((v.outbox_level, m.message_index));
+                }
+            }
+        }
+        None
```

Comment:
Perhaps try to avoid nested `if let` in favour of `?`

Reply:
Updated